api/tests/routes/backoffice_v3/offerers_test.py

```diff
@@ -1851,6 +1851,129 @@ def test_add_user_empty(self, legit_user, authenticated_client):
+            db.session.refresh(user_offerer)
+            assert user_offerer.isValidated
+            assert user_offerer.user.has_pro_role
+
+            action = history_models.ActionHistory.query.filter(
+                history_models.ActionHistory.offererId == user_offerer.offererId,
+                history_models.ActionHistory.userId == user_offerer.userId,
+            ).one()
+            assert action.actionType == history_models.ActionType.USER_OFFERER_VALIDATED
+            assert action.actionDate is not None
+            assert action.authorUserId == legit_user.id
+            assert action.userId == user_offerer.user.id
+            assert action.offererId == user_offerer.offerer.id
+            assert action.venueId is None
+
+            assert len(mails_testing.outbox) == len(user_offerers)
+
+        assert mails_testing.outbox[0].sent_data["To"] == user_offerers[0].user.email
+        assert (
+            mails_testing.outbox[0].sent_data["template"]
+            == sendinblue_template_ids.TransactionalEmail.OFFERER_ATTACHMENT_VALIDATION.value.__dict__
+        )
+
+
+class BatchOffererAttachmentRejectUnauthorizedTest(unauthorized_helpers.UnauthorizedHelperWithCsrf):
+    method = "post"
+    endpoint = "backoffice_v3_web.user_offerer_batch.user_offerer_set_batch_reject"
+    needed_permission = perm_models.Permissions.VALIDATE_OFFERER
+
+
+class BatchOffererAttachmentRejectTest:
+    def test_batch_set_offerer_attachment_reject(self, legit_user, authenticated_client):
+        user_offerers = offerers_factories.NotValidatedUserOffererFactory.create_batch(10)
+        parameter_ids = ",".join(str(user_offerer.id) for user_offerer in user_offerers)
+        url = url_for("backoffice_v3_web.user_offerer_batch.user_offerer_set_batch_reject")
+        response = send_request(
+            authenticated_client,
+            user_offerers[0].offerer.id,
+            url,
+            {"object_ids": parameter_ids, "comment": "test comment"},
+        )
+
+        assert response.status_code == 303
+        users_offerers = offerers_models.UserOfferer.query.all()
+        assert len(users_offerers) == 0
+
+        for user_offerer in user_offerers:
+            action = history_models.ActionHistory.query.filter(
+                history_models.ActionHistory.offererId == user_offerer.offererId,
+                history_models.ActionHistory.userId == user_offerer.userId,
+            ).one()
+            assert action.actionType == history_models.ActionType.USER_OFFERER_REJECTED
+            assert action.actionDate is not None
+            assert action.authorUserId == legit_user.id
+            assert action.userId == user_offerer.user.id
+            assert action.offererId == user_offerer.offerer.id
+            assert action.venueId is None
+
+            assert len(mails_testing.outbox) == len(user_offerers)
+
+        assert mails_testing.outbox[0].sent_data["To"] == user_offerers[0].user.email
```

Comment:
Par contre, quitte à tester le mail, cette partie peut être intégrée dans la boucle, en remplaçant `0` par l'index issu d'un `enumerate` placé dans le `for`.